deployments/ansible/roles/splunk-otel-collector/README.md

```diff
@@ -0,0 +1,119 @@
+
+To use this role, simply include the `splunk-otel-collector` role invocation 
+in your playbook. The following example shows how to use the role in a
+playbook with minimal required configuration:
+
+```yaml
+- name: Install Splunk OpenTelemetry Connector
+  hosts: all
+  become: yes
+  vars:
+    splunk_access_token: YOUR_ACCESS_TOKEN
+    splunk_realm: SPLUNK_REALM
+  tasks:
+    - name: "Include splunk-otel-collector"
+      include_role:
+        name: "splunk-otel-collector"
+```
+
+## Role Variables
+
+- `splunk_access_token` (**Required**): The Splunk access token to
+  authenticate requests.
+
+- `splunk_realm`: Which realm to send the data to. The `SPLUNK_REALM`
+  environment variable will be set with this value for the Splunk OTel 
+  Collector service. (**default:** `us0`)
+
+- `splunk_ingest_url`: The Splunk ingest URL, e.g.
+  `https://ingest.us0.signalfx.com`. The `SPLUNK_INGEST_URL` environment
+  variable will be set with this value for the collector service. (**default:**
+  `https://ingest.{{ splunk_realm }}.signalfx.com`)
+
+- `splunk_api_url`: The Splunk API URL, e.g. `https://api.us0.signalfx.com`.
+  The `SPLUNK_API_URL` environment variable will be set with this value for the
+  collector service. (**default:** `https://api.{{ splunk_realm }}.signalfx.com`)
+
+- `collector_version`: Version of the collector package to install, e.g.
+  `0.26.0`. (**default:** `latest`)
+
+- `collector_config`: Splunk OTel Collector config YAML file. Can be set to 
+  /etc/otel/collector/gateway_config.yaml to install the collector in gateway
+  mode. (**default:** `/etc/otel/collector/agent_config.yaml`)
+
+- `collector_config_source`: Source path to a Splunk OTel Collector config YAML 
+  file on a control node that will be uploaded and set in place of
+  `collector_config` in remote nodes. Can be used to submit a custom collector 
+  config, e.g. `./custom_collector_config.yaml`. (**default:** `""` meaning 
+  that nothing will be copied and existing `collector_config` will be used)
+
+- `splunk_bundle_dir`: The path to the [Smart Agent bundle directory](
+  https://github.com/signalfx/splunk-otel-collector/blob/main/internal/extension/smartagentextension/README.md).
+  The default path is provided by the collector package. If the specified path
+  is changed from the default value, the path should be an existing directory
+  on the node. The `SPLUNK_BUNDLE_DIR` environment variable will be set to
+  this value for the collector service.  (**default:**
+  `/usr/lib/splunk-otel-collector/agent-bundle`)
+
+- `splunk_collectd_dir`: The path to the collectd config directory for the
+  Smart Agent bundle. The default path is provided by the collector package.
+  If the specified path is changed from the default value, the path should be
+  an existing directory on the node. The `SPLUNK_COLLECTD_DIR` environment
+  variable will be set to this value for the collector service.  (**default:**
+  `/usr/lib/splunk-otel-collector/agent-bundle`)
```

Comment:
Is there a case when somebody would need to override this? I can't think of any.

Reply:
I'm not aware of any. it's just for parity with existing deployment, but I'm happy to remove it. @jcheng-splunk do you know any use-cases for that?

Reply:
If the bundle dir doesn't provide write permissions, a resource that does is required